### PR TITLE
Dvc playground tutoial [WIP]

### PR DESCRIPTION
**Pull request by sbracaloni**
Create a DVC playground tutorial

- Docker environment using compose: Git server + DVC server + user 1 + user 2
- Pipeline resources (steps and data)
- How to: 
        - Create a Pipeline
        - Reproduce a pipeline
        - Setup a remote DVC storage
        - Share with teammates
AIM: beyond the tutorial, this playground environment is useful to test how DVC works in various cases. It is a way to be more confident in data versioning.

### CODE REVIEW

File: tutorial/dvc_playground.md [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+        Running command:
+            ./steps/decrypt.py -i ./data/intermediate/file.encrypted -o ./data/intermediate/result.txt -p ./data/inputs/parameters.json
+        Saving 'data/intermediate/result.txt' to cache '.dvc/cache'.
+        Saving information to 'result.txt.dvc'.
+
+        To track the changes with git run:
+
+            git add result.txt.dvc
+
+Only the **step2**: decrypt is executed because **step1** remain up to date. `data/inputs/parameters.json` is only a dependency of **step2**.
```
Comment (SdgJlbl) [inline tutorial/dvc_playground.md]: remains

File: tutorial/dvc_playground.md [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+
+##### Revert a change
+
+You are not happy with this change, hopefully you can discard it as easily as running a git checkout.
+
+    
+    git checkout .
+    dvc checkout 
+    
+> Keep in mind to run a dvc checkout each time you run a git checkout to ensure data tracked with DVC
```
Comment (SdgJlbl) [inline tutorial/dvc_playground.md]: Keep in mind > don't forget to

File: tutorial/dvc_playground.md [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+    
+Data is stored as hard links in `.dvc/cache`. Each version of one data file is identified
+with a **md5** hash. This hash correspond to a cache entry and it changes when the data is modified.
+
+By default DVC use a local storage, but it is not really safe, you can remove it, you can easily loose
+your development server, ... And you can't share you work with your teammates.
+
+##### Setup a remote storage
+
+A lot of storage way are available using [DVC](https://dvc.org/doc/user-guide/external-outputs): Amazon S3, Google Cloud Storage, ...
```
Comment (SdgJlbl) [inline tutorial/dvc_playground.md]: storage options (?) or technologies

File: tutorial/dvc_playground.md [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+In this example we will use an **SSH** DVC server.
+
+        dvc remote add dvc_remote ssh://dvc_user@dvc_srv:/data/dvc/remote
+        dvc config core.remote dvc_remote
+        git commit -m 'Add DVC remote storage' .dvc/config
+        git push origin master
+The **remote storage** configuration is tracked using GIT.
+
+##### Push data
+Once the remote storage is set up it is possible to push data. All the data tracked with DVC is pushed to the **remote ssh storage**.
```
Comment (SdgJlbl) [inline tutorial/dvc_playground.md]: missing comma (once the remote storage is set up, )

Comment (SdgJlbl) [reply tutorial/dvc_playground.md]: Also, you use once data as singular and once as plural

File: tutorial/dvc_playground.md [hunk 5]
```diff
@@ -0,0 +1,488 @@
...
+ 
+        dvc push
+
+##### Make a mistake
```
Comment (SdgJlbl) [inline tutorial/dvc_playground.md]: Correct a mistake / Test backup

File: tutorial/dvc_playground.md [hunk 6]
```diff
@@ -0,0 +1,488 @@
...
+Now we can make a mistake and remove the whole repository loosing GIT and DVC parts on **user1** computer.
+
+        cd ..
+        rm -rf ./test_dvc_remote/
+As all code, meta and configuration files are versionned with GIT it is easy to get them back.
```
Comment (SdgJlbl) [inline tutorial/dvc_playground.md]: all the code

File: resources/dvc_playground/user/resources/steps/concat_files.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    parser.add_argument('-i', '--input-dir', required=True, help='Contains files to concat')
+    parser.add_argument('-o', '--output-file', required=True, help='Result file')
+
+    args = parser.parse_args()
+
+    if not isdir(args.input_dir):
+        logging.error(f'Not a directory: {args.input_dir}')
+    else:
+        with open(args.output_file, 'w') as fd_write:
+            for file in glob.glob(f'{args.input_dir}/*.input'):
```
[the remark this answers is not in the document]
Comment (sbracaloni) [reply resources/dvc_playground/user/resources/steps/concat_files.py]: rigth!

**Pull request merged**